Elias frontend

eliastayag opened the pull request.

eliastayag committed:
Merge branch 'master' of https://github.com/eliastayag/visipark into elias-frontend

eliastayag committed:
Cant figure out the "Add" visitor onpress function

I've set up the if statements for the card states. Default is false (which shows the Add Visitor card ui). When switched to true it switches the ui to the active visitor card. Im not sure how to go about adding the onpress function when a user clicks "Add" on the popup form.

kathleenhung5 reviewed (commented):
what you need to do now is to go into Popup.js, and add your useState function setCard1 and setCard2 in there. And of course you’ll need to figure out the how to pass your variable and function to that file/component.
The reason you need to do that is because right now when you press the button in a popup, it just closes the popup.  There’s no other function in onPress besides closing popup if you go into popup file and read that section.
After you added your setCard1 and setCard2, and the Add button is pressed, your card1 and card2 variable will be updated.  
One little note for adding your setCard1 and setCard1 in the Popup button: Make sure you use condition to determine which popup is currently showing. You’ll want to use variable “pop”  in your condition. Because otherwise it’ll add a visitor card on Visitors page no matter which popus is showing when the button is pressed.

dddora commented:
https://www.javatpoint.com/react-native-passing-value-between-screen
This is a link for passing data between two pages, hope it will help you

eliastayag committed:
attempt

eliastayag committed:
attempt

kathleenhung5 committed:
Kathleen: Review for Elias

I left some comments in Main, Tenant, Popup and Visitors. I did an example for you, hope this will help you figure out the other cases.

eliastayag committed:
Visitor Functions

Functions work for the most part but has bugs.

Bug 1 : Parking duration is working and sets the selected duration but the picker ui slides back to 2hrs everytime (like before)
Bug 2: Extend does not add to the current duration, but sets it to the selected time. (Instead of adding +2 hours to 4 hours, it sets the time to 2hours)
Bug 3: Remove button works but does not remove the correct card. Always removes the last card first. Not sure what kind of if statement to put for that.

Any tips would be helpful!
